src/colvar.cpp

```diff
@@ -2524,8 +2543,14 @@ std::string const colvar::get_state_params() const
   os << "  name " << name << "\n"
      << "  x "
      << std::setprecision(cvm::cv_prec)
-     << std::setw(cvm::cv_width)
-     << x << "\n";
+     << std::setw(cvm::cv_width);
+  if (is_enabled(f_cv_external)) {
+    // For an external colvar, x is one timestep in the future after integration
+    // write x at beginning of timestep
+    os << x_reported << "\n";
+  } else {
+    os << x << "\n";
+  }
```

Comment:
Same here as in my other comment about the addition of the new feature to `colvardeps`. "External" is a fairly generic term, which includes both quantities that are always available to Colvars (e.g. centers of mass and volmaps going through ComputeGlobal) and those like the lambda parameter that are not.  Here, this check only applies to the latter.

Reply:
Good point, this test really applies to *driven* parameters, which means
` is_enabled(f_cv_external) && is_enabled(f_cv_extended_Lagrangian)`
Updating.